haskell/private/haskell_impl.bzl

```
@@ -471,16 +484,15 @@ def haskell_library_impl(ctx):
         posix,
         dep_info,
         with_shared,
-        modules,
         exposed_modules,
         other_modules,
         my_pkg_id,
         non_empty,
     )
 
     interface_dirs = depset(
-        direct = c.interface_files + [m[HaskellModuleInfo].interface_file for m in modules],
-        transitive = [dep_info.interface_dirs],
+        direct = [f for f in c.interface_files],
```

```suggestion
        direct = c.interface_files,
```
Redundant list comprehension.

---

Redundant, yes. Except that without it, bazel will complain that the list is mutable. What's is the right fix here?

---

Oh, yes, that's annoying. Would `list(c.interface_files)` work? Otherwise, happy to keep the comprehension

---

I removed this and couldn't reproduce the error. So it is gone for now.